### PR TITLE
add support for tcp_keepalive, tcp_keepidle, tcp_keepintvl options

### PR DESCRIPTION
**Pull request by sewenthy**
These options help with sending TCP level probes to make sure connection is alive.  

See: https://curl.se/libcurl/c/CURLOPT_TCP_KEEPALIVE.html

### CODE REVIEW

File: curl.ml [hunk 1]
```diff
@@ -494,6 +494,9 @@ type curlOption =
   | CURLOPT_XFERINFOFUNCTION of (int64 -> int64 -> int64 -> int64 -> bool)
   | CURLOPT_PREREQFUNCTION of (string -> string -> int -> int -> bool)
   | CURLOPT_AWS_SIGV4 of string
+  | CURLOPT_TCP_KEEPALIVE of bool
+  | CURLOPT_TCP_KEEPIDLE of int64
```
Comment (ygrek) [inline curl.ml:498]: int64 is not convenient, int is enough

**Review by ygrek — CHANGES_REQUESTED** [review]
missing actual implementation/mapping of these options on C side in curl-helper.c

File: curl.ml (hunk 1 above)
Comment (sewenthy) [reply curl.ml]: thanks! addressed in https://github.com/ygrek/ocurl/pull/70/commits/d9bdf57bac1b08eeec278772489c6fad620f66dd

File: curl-helper.c [hunk 2]
```diff
@@ -3326,6 +3326,17 @@ SETOPT_LONG( BUFFERSIZE)
   SETOPT_STRING( AWS_SIGV4)
 #endif
 
+#if HAVE_DECL_CURLOPT_TCP_KEEPALIVE
+  SETOPT_BOOL( TCP_KEEPALIVE)
+#endif
+
+#if HAVE_DECL_CURLOPT_TCP_KEEPIDLE
+  SETOPT_INT( TCP_KEEPIDLE)
```
Comment (ygrek) [inline curl-helper.c:3334]: did you try compiling? there is no SETOPT_INT
and then next question - was it tested?

File: curl-helper.c [hunk 3]
```diff
@@ -3766,6 +3777,21 @@ CURLOptionMapping implementedOptionMap[] =
 #else
   HAVENOT(AWS_SIGV4),
 #endif
+#if HAVE_DECL_CURLOPT_TCP_KEEPALIVE
```
Comment (ygrek) [inline curl-helper.c:3780]: there is no need for configure test btw, these options were added in 7.25.0, ocurl requires 7.28.0 at minimum, can just add unconditionally.

**Review by ygrek — CHANGES_REQUESTED** [review]

File: curl-helper.c (hunk 2 above)
Comment (sewenthy) [reply curl-helper.c]: once i removed the conditional add `#if HAVE_DECL_CURLOPT_TCP_KEEPIDLE`, the compilation failed but with it the compile actually passed for me--might be some macro magic i'm not too sure.

i did a small test with a python server with the setup and i do see extra TCP requests dumped (when there wasn't any without the setup):
```
13:56:31.573793 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [S], seq 2756318970, win 65535, options [mss 65495,sackOK,TS val 3733098363 ecr 0,nop,wscale 14], length 0
E..<<i@.@..Q.............J...........0.........
...{........
13:56:31.573845 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [.], ack 663868098, win 4, options [nop,nop,TS val 3733098363 ecr 3733098363], length 0
E..4<j@.@..X.............J..'........(.....
...{...{
13:56:31.573921 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [P.], seq 0:117, ack 1, win 4, options [nop,nop,TS val 3733098363 ecr 3733098363], length 117
E...<k@.@................J..'..............
...{...{GET / HTTP/1.1
Host: 127.0.0.1:3232
User-Agent: Curl_lwt
Accept: */*
Accept-Encoding: deflate, gzip, br, zstd


13:56:32.576286 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [.], ack 1, win 4, options [nop,nop,TS val 3733099365 ecr 3733098363], length 0
E..4<l@.@..V.............J.o'........(.....
...e...{
13:56:35.616246 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [.], ack 1, win 4, options [nop,nop,TS val 3733102405 ecr 3733099366], length 0
E..4<m@.@..U.............J.o'........(.....
...E...f
13:56:36.310771 IP ip6-localhost.40152 > ip6-localhost.3232: Flags [R], seq 3376754125, win 0, length 0
E..(..@.@.<..............E-.....P....S..
13:56:38.688243 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [.], ack 1, win 4, options [nop,nop,TS val 3733105477 ecr 3733102405], length 0
E..4<n@.@..T.............J.o'........(.....
...E...E
13:56:41.579093 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [F.], seq 117, ack 1, win 4, options [nop,nop,TS val 3733108368 ecr 3733105477], length 0
E..4<o@.@..S.............J.p'........(.....
.......E
13:56:46.321807 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [R], seq 2756319089, win 0, length 0
E..(..@.@.<..............J.q....P...L...
```

**Comment by sewenthy** [comment]
fixup in: https://github.com/ygrek/ocurl/pull/70/commits/170985041a1bd7fa1d629352e095a6e2d5153986
test in: https://github.com/ygrek/ocurl/pull/70/commits/95e6ff8643581c261cf2f5f85e80b02fdf8b72a4

thanks!

**Review by ygrek — APPROVED** [review]

**Pull request merged**